web/src/helpers.ts

```diff
@@ -0,0 +1,145 @@
+export function filterSet<T>(
```

Comment:
I think this whole file may not be necessary. Try removing it and see if everything still works. If yes, remove it.